test/system/dogstatsd/receive_and_forward_test.go

```diff
@@ -40,34 +42,50 @@ func testMetadata(t *testing.T, d *dogstatsdTest) {
 }
 
 func TestReceiveAndForward(t *testing.T) {
-	d := setupDogstatsd(t)
-	defer d.teardown()
-	defer log.Flush()
 
-	testMetadata(t, d)
+	tests := map[string]struct {
+		kind string
+	}{
+		"zlib": {kind: compression.ZlibKind},
+		"zstd": {kind: compression.ZstdKind},
+	}
 
-	d.sendUDP("_sc|test.ServiceCheck|0")
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			d := setupDogstatsd(t)
+			defer d.teardown()
+			defer log.Flush()
 
-	timeOut := time.Tick(30 * time.Second)
-	select {
-	case <-d.requestReady:
-	case <-timeOut:
-		require.Fail(t, "Timeout: the backend never receive a requests from dogstatsd")
-	}
+			testMetadata(t, d)
 
-	requests := d.getRequests()
-	require.Len(t, requests, 1)
+			d.sendUDP("_sc|test.ServiceCheck|0")
+
+			timeOut := time.Tick(30 * time.Second)
+			select {
+			case <-d.requestReady:
+			case <-timeOut:
+				require.Fail(t, "Timeout: the backend never receive a requests from dogstatsd")
+			}
 
-	sc := []servicecheck.ServiceCheck{}
-	decompressedBody, err := compression.Decompress([]byte(requests[0]))
-	require.NoError(t, err, "Could not decompress request body")
-	err = json.Unmarshal(decompressedBody, &sc)
-	require.NoError(t, err, fmt.Sprintf("Could not Unmarshal request body: %s", decompressedBody))
+			requests := d.getRequests()
+			require.Len(t, requests, 1)
 
-	require.Len(t, sc, 2)
-	assert.Equal(t, sc[0].CheckName, "test.ServiceCheck")
-	assert.Equal(t, sc[0].Status, servicecheck.ServiceCheckOK)
+			mockConfig := pkgconfigsetup.Conf()
```

Comment:
same here it sounds strange not to see tabs.

Reply:
Strange, it shows as tabs in my editor. 🤔